Stb (#3823)
mlpack/mlpack: HISTORY.md

 * Add `MLPACK_NO_STD_MUTEX` to allow disabling `std::mutex` (#3868).

>  * Make STB part of mlpack and integrate `ResizeImages` functionality (#3823).

Reviewer: ```suggestion
 * Bundle STB with mlpack and add `ResizeImages()` functionality (#3823).
```

Up to you, I think it's a little clearer but either way is fine.
